PWGCF/FEMTOSCOPY/FemtoDream/AliAnalysisTaskNanoPt.cxx

```diff
@@ -0,0 +1,424 @@
+   fAntiDeuteronList = fAntiDeuteronTrack->GetQAHists();
+   fAntiDeuteronList->Add(fAntiDeuteronRestMass);
+
+  fResultsQA = new TList();
+  fResultsQA->SetOwner();
+  fResultsQA->SetName("ResultsQA");
+
+  if (fConfig->GetUseEventMixing()) 
+  {
+    fResults = fPartColl->GetHistList();
+    if (!fConfig->GetMinimalBookingME()) 
+    {
+      fResultsQA->Add(fPartColl->GetQAList());
+      fResultsQA->Add(fPairCleaner->GetHistList());
+    }
+  }
+  else 
+  {
+    fResults = new TList();
+    fResults->SetOwner();
+    fResults->SetName("Results");
+  }
+
+  PostData(1, fEvtList);
+  PostData(2, fProtonList);
+  PostData(3, fAntiProtonList);
+  PostData(4, fDeuteronList);
+  PostData(5, fAntiDeuteronList);
+  PostData(6, fResults);
+  PostData(7, fResultsQA);
+
+}
+
+//------------------------------------------UserExec()----------------------------------------------------------------------------
+
+
+void AliAnalysisTaskNanoPt::UserExec(Option_t  *option ) {
+  AliVEvent *fInputEvent = InputEvent();
+  if (fIsMC)fMCEvent = MCEvent();
+
+  // PREAMBLE - CHECK EVERYTHING IS THERE
+  if (!fInputEvent) {
+    AliError("No Input event");
+    return;
+  }
+
+  if (fIsMC && !fMCEvent) {
+    AliError("No MC event");
+    return;
+  }
+
+  if (!fEvtCuts) {
+    AliError("Event Cuts missing");
+    return;
+  }
+
+  if (!fProtonTrack || !fAntiProtonTrack) {
+    AliError("Proton Cuts missing");
+    return;
+  }
+
+  if (!fDeuteronTrack || !fAntiDeuteronTrack) {
+    AliError("Deuteron Cuts missing");
+    return;
+  }
+
+
+ 
+
+  // EVENT SELECTION
+  fEvent->SetEvent(fInputEvent);
+  if (!fEvtCuts->isSelected(fEvent))
+    return;
+
+
+  // PROTON SELECTION
+  ResetGlobalTrackReference();
+  for (int iTrack = 0; iTrack < fInputEvent->GetNumberOfTracks(); ++iTrack) {
+    AliVTrack *track = static_cast<AliVTrack *>(fInputEvent->GetTrack(iTrack));
+    if (!track) {
+      AliFatal("No Standard AOD");
+      return;
+    }
+
+    StoreGlobalTrackReference(track);
+
+ }
+  std::vector<AliFemtoDreamBasePart> Proton;
+  std::vector<AliFemtoDreamBasePart> AntiProton;
+  std::vector<AliFemtoDreamBasePart> Deuteron;
+  std::vector<AliFemtoDreamBasePart> AntiDeuteron;
+  
+  const int multiplicity = fEvent->GetMultiplicity();
+  fTrack->SetGlobalTrackInfo(fGTI, fTrackBufferSize);
+
+  for (int iTrack = 0; iTrack < fInputEvent->GetNumberOfTracks(); ++iTrack) {
+    AliVTrack *track = static_cast<AliVTrack *>(fInputEvent->GetTrack(iTrack));
+    fTrack->SetTrack(track, fInputEvent, multiplicity);
+    fTrack->SetInvMass(0.938);
+
+    if (fProtonTrack->isSelected(fTrack)) 
+    {
+      
+      fProtonRestMass->Fill(fTrack->GetPt(), GetMass2sq(fTrack));
+      Proton.push_back(*fTrack);
+    }
+
+    if (fAntiProtonTrack->isSelected(fTrack)) 
+    {
+      
+      fAntiProtonRestMass->Fill(fTrack->GetPt(), GetMass2sq(fTrack));
+       AntiProton.push_back(*fTrack);
+    }
+    if (fDeuteronTrack->isSelected(fTrack)) 
+
```

Comment:
To improve the readability consider to use a code formatting tool (your editor should have one), otherwise these empty lines might get confusing